src/logger.ts

```diff
@@ -1,26 +1,44 @@
 import pc from 'picocolors'
+import ora from 'ora'
 
 const defaultColorFn = (text: string) => text
 function color(prefixColor: any) {
   return pc.isColorSupported ? (pc as any)[prefixColor] : defaultColorFn
 }
 
+export const buildingSpinner = ora({
+  text: 'bundling...\n\n',
+  spinner: 'dots',
+  color: 'green',
+})
+
+function clearSpinner() {
+  if (buildingSpinner.isSpinning) {
+    buildingSpinner.stop()
+  }
+}
+
 export const logger = {
   log(...arg: any[]) {
+    clearSpinner()
```

Comment:
logging should be isolated from loading, otherwise every time you log sth it will accidentally stop the spinner

Reply:
Yes, so I asked do we need to restore the spinner after the new log output. It seems the only way we can make ora doesn't swallow stdout https://github.com/sindresorhus/ora/issues/90. This is also how vitejs does https://github.com/vitejs/vite/pull/269/files.